### ISSUE TITLE
[Feature Request] Allow RBAC based on node_labels with same key with multiple values

### ISSUE THREAD

**Issue by ken5scal** [gravitational/teleport#2136]
**What happened**: This is an issue to request a feature.

**What you expected to happen**: 
* What we want: Allowing to declare node_labels which permits certain role to login nodes which share same key but different values.

For example, say there are two nodes each have a label "application -> web01"  or "application -> web02".  We would want  a 'web' role to be able to login both nodes; but
it is not available yet  (correct me if I'm wrong) , because `spec.allow.node_labels` do not allow to declare in a such way.

```
kind: role
version: v3
metadata:
  name: web
spec:
  allow:
    logins: [root, '{{internal.logins}}']

    # node labels that a user can connect to. The wildcard ('*') means "any node"
    # Following is not Possible★★
    # node_labels:
    #   application: web0\d   <-  ★★regex is not supported

    # node_labels:
    #   application: web01   <-  ★★node_labels with same key w/ multiple values are not supported
    #   application: web02   

   # node_lables:
   #      application:      <- ★★ this is not supported as well.
   #          - web01
   #          - web02
```
So either implementing regex expression, the map[label_key][]label_value could be nice for us.

**How to reproduce it (as minimally and precisely as possible)**:
n/a

**Environment**:
- Teleport version (use `teleport version`): Teleport Enterprise v2.6.7 git:v2.6.7-0-g280f0fe3
- Tsh version (use `tsh version`): Teleport v2.6.7 git:v2.6.7-0-g280f0fe3
- OS (e.g. from /etc/os-release): N/A

**Browser environment**
N/A
**Relevant Debug Logs If Applicable**
N/A

**Comment by klizhentas** [comment]
I think node_labels as a list not as a map is a good idea @kontsevoy what do you think?

**Comment by klizhentas** [comment]
The upside of this approach in addition to being flexible it does not include any migrations or structure changes, so it will be easy to add.

**Comment by kontsevoy** [comment]
@klizhentas  love it.

**Comment by klizhentas** [comment]
ok, for the record, we have decided to support both formats:

scalar values with regexp

```yaml
allow:
  node_labels:
    'env: '^production|staging$'
deny:
  node_labels:
    'env: '^test$'
```

list values:

```yaml
allow:
  node_labels:
    'env: ['^production|staging$', 'test']
deny:
  node_labels:
    'env: ['^test$', 'list']
```

**Comment by klizhentas** [comment]
k, I'm going to close this one as it has been partially implemented and create a new one just for regular expressions.

**Comment by klizhentas** [comment]
It's going to be available in the next 2.7.x release upgrade this week.

**Comment by ken5scal** [comment]
tx!
